js/dateProcessor.js

```diff
@@ -0,0 +1,68 @@
+var dateProcessor = {
```

Comment:
There is a rule for this keywords: If you don't need var, use let instead. If you don't need let, use const instead. Really helps when it comes to assigning values to variables.